### DISCUSSION TITLE
Updating advertisement data

### DISCUSSION THREAD

**Discussion by homy-newfs8** [h2zero/NimBLE-Arduino#827] [Q&A] [answered]
NimBLEAdvertising::setName(), setManufacturerData() (and the corresponding methods in NimBLEAdvertisementData) seem to append data.
If I want to update some of the advertising data, I have to start with clearData() and set all the data again.
Is this an intentional design?

**Comment by h2zero** [comment, marked as answer]
@homy-newfs8 Yes, this is intentional because many fields are variable length so updating just one thing in the data would introduce too much complexity.

**Comment by homy-newfs8** [comment]
I see, thanks.
